clients/drcachesim/tests/record_filter_unit_tests.cpp

```diff
@@ -321,6 +321,20 @@ test_encodings2regdeps_filter()
         { { TRACE_TYPE_ENCODING, 4, { 0xe78948 } }, true, { false } },
         { { TRACE_TYPE_ENCODING, 8, { 0x0006090600010011 } }, false, { true } },
         { { TRACE_TYPE_INSTR, 3, { 0x7f6fdd3ec360 } }, true, { true } },
+        // Same instr same chunk.
+        { { TRACE_TYPE_INSTR, 3, { 0x7f6fdd3ec360 } }, true, { true } },
+        // Unit header.
+        { { TRACE_TYPE_MARKER, TRACE_MARKER_TYPE_TIMESTAMP, { 0xc } }, true, { true } },
+        { { TRACE_TYPE_MARKER, TRACE_MARKER_TYPE_CPU_ID, { 0xd } }, true, { true } },
+        // Duplicated encoding across chunk boundary.
```

Comment:
There's no chunk boundary here: I see only a timestamp+cpuid.  You need a chunk instr count, chunk footer, and record ordinal markers.  See test_chunk_update().

Reply:
A chunk here is a compress storage (zipfile) subcomponent allowing fast seeking.

Reply:
Chunk boundaries must not only be at precise instruction counts but include record count hints, so the record_filter framework has to update them for pretty much any filter doing anything.

Reply:
So the bug had nothing to do with record_filter and its encodings2regdeps filter, it was in the test case.
Specifically, I wasn't setting `write_archive=true` in the record_filter constructor, so that's why it wasn't generating new chunks.
The test case has been fixed and now it passes (I'm confident the expected output is correct, and I talked to @abhinav92003  about it as well, but an additional pair of eyes to check that doesn't hurt).